src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -171,17 +171,15 @@ protected override ImmutableArray<LabelSymbol> BuildLabels()
             // part of the label, but we do not report any diagnostics here. Diagnostics will be reported during binding.
 
             ArrayBuilder<LabelSymbol> labels = ArrayBuilder<LabelSymbol>.GetInstance();
-            DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
             foreach (var section in SwitchSyntax.Sections)
             {
                 // add switch case/default labels
-                BuildSwitchLabels(section.Labels, GetBinder(section), labels, tempDiagnosticBag);
+                BuildSwitchLabels(section.Labels, GetBinder(section), labels, BindingDiagnosticBag.Discarded);
```

Comment:
>BindingDiagnosticBag.Discarded [](start = 78, length = 30)

Consider declaring this above where the temp bag was created before, as this API allocates every time it is called.

Reply:
Ah, wait, this the `Discarded` on the other `BindingDiagnosticBag`. Another reason to rename.

---
In reply to: [361556164](https://github.com/dotnet/roslyn/pull/40560#discussion_r361556164) [](ancestors = 361556164)